src/lib/Persistence/Cache/ContentHandler.php

```diff
@@ -433,16 +436,68 @@ public function addRelation(RelationCreateStruct $relation)
     {
         $this->logger->logCall(__METHOD__, ['struct' => $relation]);
 
+        $this->cache->invalidateTags([
+            $this->cacheIdentifierGenerator->generateTag(
+                self::CONTENT_IDENTIFIER,
+                [$relation->destinationContentId]
+            ),
+        ]);
+
         return $this->persistenceHandler->contentHandler()->addRelation($relation);
     }
 
     /**
      * {@inheritdoc}
      */
-    public function removeRelation($relationId, $type)
+    public function removeRelation($relationId, $type, ?int $destinationContentId = null)
```

Comment:
nitpick: you can safely add `:void` here as this is our internal implementation and we have also control over the interface.
```suggestion
    public function removeRelation($relationId, $type, ?int $destinationContentId = null): void
```